docs_src/design/adr/0023-North-South-Messaging.md

```diff
@@ -0,0 +1,527 @@
+
+Since most message bus protocols lack a generic message header mechanism (as in HTTP), providing request/response metadata is accomplished by defining a `message` envelope object associated with each request/response.  Therefore, messages described in this ADR must provide JSON `envelope` and `payload` objects for each request/response.
+
+The message topic names act like the HTTP paths and methods in REST requests.  That is, the topic names specify the device receiver of any command request as paths do in the HTTP requests.
+
+![image](command-msg-structure.png)
+
+#### Message Envelope
+
+The messages defined in this ADR are JSON formatted requests and responses that share a common base structure. The outer most JSON object represents the message `envelope`, which is used to convey metadata about request/response (e.g. a correlation identifier which will be added to any relayed request message as well as the response message envelope so that the 3rd party system will know to associate the responses to the original request).
+
+!!! Note
+        A Correlation ID (see this [article](https://www.rapid7.com/blog/post/2016/12/23/the-value-of-correlation-ids/) for a more detailed description) is a unique value that is added to every request and response involved in a transaction which could include multiple requests/responses between one or more microservices. It's not meant to correlate requests to responses, its meant to label every message involved in a potentially multi-request transaction.
+
+        A Request ID should be an identifier returned on the response to a request (providing traceability between single request/response).
+
+The `envelope` will also contain the API version (something provided in the HTTP path when using REST).
+
+Command requests in HTTP may also contain ds-pushevent and ds-returnevent query parameters (for GET commands).  These will be optionally provided key/value pairs represented in the message `envelope`'s query parameters (and optionally allows for other parameters in the future).
+
+``` JSON
+{
+    "Correlation-ID": "14a42ea6-c394-41c3-8bcd-a29b9f5e6835",
+    "API":"V2",
+    "queryParams": {
+        "ds-pushevent":"yes",
+        "ds-returnevent":"yes",
+     }
+     ...
+}
+```
+
+#### Command Message Payload
+
+The **request** message `payload` to the command service and those relayed to the device service would mimic their HTTP/REST request body alternatives.  The `payload` provides details needed in executing the command at the south side.
+
+In the example GET and PUT messages below, note the `envelope` wraps or encases the message `payload`.  The payload may be empty (as is typical of GET requests).
```

Comment:
Why include the `payload` if it's empty? Why not just make it optional element?

Reply:
pulled the payload and added a note that it was optional.